sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String containerName, String blobName, AccessTier accessTier) {
+        return setTierHelper(String.format("%s/%s", containerName, blobName), accessTier, null);
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-String-AccessTier-LeaseAccessConditions}
+     *
+     * @param containerName The container of the blob.
+     * @param blobName The name of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @param leaseAccessConditions Lease access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String containerName, String blobName, AccessTier accessTier,
+        LeaseAccessConditions leaseAccessConditions) {
+        return setTierHelper(String.format("%s/%s", containerName, blobName), accessTier, leaseAccessConditions);
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-AccessTier}
+     *
+     * @param blobUrl URI of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String blobUrl, AccessTier accessTier) {
+        return setTierHelper(getUrlPath(blobUrl), accessTier, null);
+    }
+
+    /**
+     * Adds a set tier operation to the batch.
+     *
+     * <p><strong>Code sample</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobBatch.setTier#String-AccessTier-LeaseAccessConditions}
+     *
+     * @param blobUrl URI of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @param leaseAccessConditions Lease access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setTier(String blobUrl, AccessTier accessTier, LeaseAccessConditions leaseAccessConditions) {
```

Comment:
>Response<Void> [](start = 11, length = 14)

I think I also left this on the other PR, but Mono<Response> makes more sense to me here as the operation isn't completed until the batch is submitted. Mono.create should give you a way to fill in the value of the mono once the batch has been submitted and processed.

Reply:
One additional thought about why Mono is preferred is because otherwise you can't know for sure when the batch has finished and will be stuck polling. Your comment about the error from assertResponseReceived in the BatchResponse (or whatever that check is called) is recoverable by actually submitting the batch, but then there's no way of knowing when it is safe to access the response after that. Just because you call submit batch, doesn't mean the response will be immediately filled and ok to access. 

---
In reply to: [333123167](https://github.com/Azure/azure-sdk-for-java/pull/5734#discussion_r333123167) [](ancestors = 333123167)

Reply:
This can be changed in preview 5 if we find this pattern to be bad